tripviewer/web/Views/Trip/Index.cshtml

```diff
@@ -1,119 +1,145 @@
+            directionsManager = new Microsoft.Maps.Directions.DirectionsManager(map);
+
+            @*var startpoint = @($"{Model.FirstOrDefault().Latitude},{Model.FirstOrDefault().Longitude}");
+            var endpoint = @($"{Model.Last().Latitude},{Model.Last().Longitude}");*@
+
+            //Create waypoints to route between.
+            var startWaypoint = new Microsoft.Maps.Directions.Waypoint({ address: '@($"{Model.FirstOrDefault().Latitude},{Model.FirstOrDefault().Longitude}")' });
+            directionsManager.addWaypoint(startWaypoint);
+
+            var endWaypoint = new Microsoft.Maps.Directions.Waypoint({ address: '@($"{Model.Last().Latitude},{Model.Last().Longitude}")' });
+            directionsManager.addWaypoint(endWaypoint);
+
+            //Add event handler to directions manager.
+            Microsoft.Maps.Events.addHandler(directionsManager, 'directionsUpdated', directionsUpdated);
+
+            //Calculate directions.
+            directionsManager.calculateDirections();
+        });
+    }
+
+    function directionsUpdated(e) {
+        //Remove any previously calculated locations from the map.
+        map.entities.clear();
+
+        //Get the current route index.
+        var route = directionsManager.getCurrentRoute();
+
+        if (route && route.routePath && route.routePath.length > 0) {
+
+            //Create an array to store the calculated locations, add the starting location.
+            var locationsAlongPath = [route.routePath[0]];
+
+            //Calculate the length of the route.
+            var routeLength = Microsoft.Maps.SpatialMath.getLengthOfPath(route.routePath, Microsoft.Maps.SpatialMath.DistanceUnits.Miles)
+
+            var numSteps = Math.floor(routeLength / stepDistance);
+            var loc;
+
+            for (var i = 1; i <= numSteps; i++) {
+                loc = Microsoft.Maps.SpatialMath.getLocationAlongPath(route.routePath, stepDistance * i, Microsoft.Maps.SpatialMath.DistanceUnits.Miles);
+                locationsAlongPath.push(loc);
+            }
+
+            //Add the last location on the route.
+            locationsAlongPath.push(route.routePath[route.routePath.length - 1]);
+
+            //Do something with the calculates locations. Lets show red pushpins for now.
+            //for (var i = 0, len = locationsAlongPath.length; i < len; i++) {
+            //    map.entities.push(new Microsoft.Maps.Pushpin(locationsAlongPath[i], { color: 'red' }))
+            //}
+        }
+        }
+
+        function showDetail() {
+            var x = document.getElementById("myInfoPanel");
+            if (x.style.display === "none") {
+                x.style.display = "block";
+            } else {
+                x.style.display = "none";
+            }
+        }
+    </script>
+    <script type='text/javascript' src='https://www.bing.com/api/maps/mapcontrol?callback=GetMap&key=Ar6iuHZYgX1BrfJs6SRJaXWbpU_HKdoe7G-OO9b2kl3rWvcawYx235GGx5FPM76O' async defer></script>
```

Comment:
This needs to be extracted into ENV var.  For now, let's just change the secret with a ###changeme###.